### PR TITLE
FIX: Show error message if no uploads are allowed

### PR DESCRIPTION
**Pull request by nbianca**
It used to fail without displaying an error message if no upload extensions were authorized. This also disables the button in the first place to avoid displaying an error to the user (the error will be displayed only when drag and dropping a file).

Screenshot 2022-11-29 at 16 27 13

### CODE REVIEW

File: app/assets/javascripts/discourse/tests/unit/lib/uploads-test.js [hunk 1]
```diff
@@ -138,6 +138,38 @@ discourseModule("Unit | Utility | uploads", function () {
     );
   });
 
+  test("shows error message when no extensions are authorized", function (assert) {
+    this.siteSettings.authorized_extensions = "";
+    this.siteSettings.authorized_extensions_for_staff = "";
```
Comment (tgxworld) [inline app/assets/javascripts/discourse/tests/unit/lib/uploads-test.js:143]:
```suggestion
    this.siteSettings.authorized_extensions = "";
```

I think we only need to change this site setting for this test case.

File: app/assets/javascripts/discourse/tests/unit/lib/uploads-test.js [hunk 2]
```diff
@@ -138,6 +138,38 @@ discourseModule("Unit | Utility | uploads", function () {
...
+      })
+    );
+    assert.ok(
+      dialog.alert.calledWith(I18n.t("post.errors.no_uploads_authorized"))
+    );
+  });
+
+  test("shows error message when no extensions are authorized for staff", function (assert) {
+    this.siteSettings.authorized_extensions = "";
+    this.siteSettings.authorized_extensions_for_staff = "";
```
Comment (tgxworld) [inline app/assets/javascripts/discourse/tests/unit/lib/uploads-test.js:159]:
```suggestion
    this.siteSettings.authorized_extensions_for_staff = "";
```

I think we only need to change one site setting for this test case.

File: config/locales/client.en.yml [hunk 3]
```diff
@@ -3280,6 +3280,7 @@ en:
           one: "Sorry, you can only upload %{count} file at a time."
           other: "Sorry, you can only upload %{count} files at a time."
         upload_not_authorized: "Sorry, the file you are trying to upload is not authorized (authorized extensions: %{authorized_extensions})."
+        no_uploads_authorized: "Sorry, no files are allowed to be uploaded."
```
Comment (tgxworld) [inline config/locales/client.en.yml:3283]: Looking at this description, I wonder if this error is even useful. Should we just disable the upload button completely if there are not authorized extensions? Kind of odd that we're telling the user that no files are allowed to be uploaded after they have uploaded.

File: app/assets/javascripts/discourse/tests/unit/lib/uploads-test.js (hunk 2 above)
Comment (nbianca) [reply app/assets/javascripts/discourse/tests/unit/lib/uploads-test.js:158]: Both site settings are taken into account because the test user is staff.

**Review by tgxworld — APPROVED** [review]
Changes look good to me but I do have a question about the usefulness of the error message when no uploads have been authorised. Is there an internal discussion somewhere we can discuss about this?

approving first because I don’t think my question needs to block the changes here in case I’m missing more context.

**Pull request merged**